index.js

```diff
@@ -45,11 +45,27 @@ function tryRequire(id) {
 	}
 }
 
+const errorRegExp = /Package subpath '.\/package.json' is not defined by "exports" in (.*)/;
+function tryParseErrorMessage(error) {
+	try {
+		const [, file] = errorRegExp.exec(error.message);
+		return file;
+	} catch (err) {
+		// if the matching fails, rethrow the original error for inspection
+		throw error;
+	}
+}
+
 function tryResolve(pkg, importer) {
 	try {
 		return relative.resolve(pkg, importer);
 	} catch (err) {
 		if (err.code === 'MODULE_NOT_FOUND') return null;
+		if (err.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED')
+			// in the edge cases where the package.json file is not listed in the
+			// package's "exports" field, parse the target file name from the error
+			// message
+			return tryParseErrorMessage(err);
```

Comment:
I don't think that parsing the error message is a good idea. I think the error message is nothing one could rely on, it could change at any point in time.

Instead, I believe that right now the plugin should probably just emit a warning if it cannot resolve the `package.json` so that package authors can start exporting it if necessary (it's not always necessary, in the case of `uuid` it is definitely not necessary for the svelte plugin to read `uuid`'s `package.json` since it doesn't contain any svelte-specific information).

See: https://github.com/uuidjs/uuid/issues/444#issuecomment-630424195

We might even see Node.js change it's behavior again and always make `package.json` resolvable, see https://github.com/nodejs/node/issues/33460